bigcache.go

```diff
@@ -33,6 +33,79 @@ type cacheShard struct {
 	onRemove    func(wrappedEntry []byte)
 }
 
+// EntryInfo holds informations about entry in the cache
+type EntryInfo struct {
```

Comment:
So whats whit this? I did a small benchmark and gready extraction is a little bit slower than copy.
<details>
```diff
diff --git a/bigcache.go b/bigcache.go
index d610216..f15c1de 100644
--- a/bigcache.go
+++ b/bigcache.go
@@ -35,27 +35,30 @@ type cacheShard struct {
 
 // EntryInfo holds informations about entry in the cache
 type EntryInfo struct {
-       entry []byte
+       timestamp uint64
+       hash      uint64
+       key       string
+       value     []byte
 }
 
 // Key returns entry's underlying key
 func (e EntryInfo) Key() string {
-       return readKeyFromEntry(e.entry)
+       return e.key
 }
 
 // Hash returns entry's hash value
 func (e EntryInfo) Hash() uint64 {
-       return readHashFromEntry(e.entry)
+       return e.hash
 }
 
 // Timestamp returns entry's timestamp (time of insertion)
 func (e EntryInfo) Timestamp() uint64 {
-       return readTimestampFromEntry(e.entry)
+       return e.timestamp
 }
 
 // Value returns entry's underlying value
 func (e EntryInfo) Value() []byte {
-       return readEntry(e.entry)
+       return e.value
 }
 
 // EntryInfoIterator allows to iterate over entries in the cache
@@ -140,11 +143,11 @@ func (it *EntryInfoIterator) Value() (EntryInfo, error) {
                return EntryInfo{}, fmt.Errorf("Could not retrieve entry from cache")
        }
 
-       var dst = make([]byte, len(entry))
-       copy(dst, entry)
-
        return EntryInfo{
-               entry: dst,
+               timestamp: readTimestampFromEntry(entry),
+               hash:      readHashFromEntry(entry),
+               key:       readKeyFromEntry(entry),
+               value:     readEntry(entry),
        }, nil
 }
```

```shell
$go test ./... -bench=BenchmarkIterateOverCache -run=NONE -count 3 -cpu 1,6 -benchmem > new.txt && git stash && go test ./... -bench=BenchmarkIterateOverCache -run=NONE -count 3 -cpu 1,6 benchmem > old.txt 
$ benchcmp old.txt new.txt                                                                                                                                                                                     
benchmark                                                    old ns/op     new ns/op     delta
BenchmarkIterateOverCache/512-shards        224           224           +0.00%
BenchmarkIterateOverCache/512-shards        217           223           +2.76%
BenchmarkIterateOverCache/512-shards        218           248           +13.76%
BenchmarkIterateOverCache/512-shards-6      273           293           +7.33%
BenchmarkIterateOverCache/512-shards-6      273           284           +4.03%
BenchmarkIterateOverCache/512-shards-6      274           284           +3.65%
BenchmarkIterateOverCache/1024-shards       214           216           +0.93%
BenchmarkIterateOverCache/1024-shards       214           215           +0.47%
BenchmarkIterateOverCache/1024-shards       215           218           +1.40%
BenchmarkIterateOverCache/1024-shards-6     268           279           +4.10%
BenchmarkIterateOverCache/1024-shards-6     271           278           +2.58%
BenchmarkIterateOverCache/1024-shards-6     268           281           +4.85%
BenchmarkIterateOverCache/8192-shards       209           210           +0.48%
BenchmarkIterateOverCache/8192-shards       207           210           +1.45%
BenchmarkIterateOverCache/8192-shards       207           211           +1.93%
BenchmarkIterateOverCache/8192-shards-6     265           276           +4.15%
BenchmarkIterateOverCache/8192-shards-6     266           275           +3.38%
BenchmarkIterateOverCache/8192-shards-6     263           279           +6.08%
```
</details>